regression/test.pl

```diff
@@ -115,12 +112,44 @@ ($$$$$)
         last if($included == -1);
         if($result eq "--") {
           $included--;
-        } else {
+        } elsif($result !~ /\s*/) {
           my $r;
-          $result =~ s/\\/\\\\/g;
-          $result =~ s/([^\\])\$/$1\\r\\\\?\$/;
-          system("bash", "-c", "grep $grep_options \$'$result' '$name/$output' >/dev/null");
-          $r = ($included ? $? != 0 : $? == 0);
+
+          my $dir = getcwd();
+          my $abs_path = "$dir/$name/$output";
+          open( my $fh => $abs_path) || die "Cannot open '$name/$output': $!";
+
+          # Multi line therefore we run each check across the whole output
+          if($grep_options eq "activate-multi-line-match") {
+            local $/ = undef;
+            binmode $fh;
+            my $whole_file = <$fh>;
+            my $is_match = $whole_file =~ /$result/;
+            $r = ($included ? !$is_match : $is_match);
+          }
+          else
```

Comment:
My vote is for the Perl solution since it (appears to) work with all the existing tests that are covered by `make -c regressions`. Further, in the cases where the two syntax don't match up (which I can't think of any since both are Perl flavoured Regex) I think it would be confusing to test writers and maintainers if the format depended on whether it was multi line or not.